<a href="https://colab.research.google.com/github/ravirude/Project_Portfolio/blob/main/Netflix_Exploratory_Analysis_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Netflix Exploratory Data Analysis

Introduction.

This is Exploratory Data Analysis using Python of the "Netflix Movies & TV Shows" dataset. The purpose of this project is to find out and visualize the data's main characteristics and trends using statistical methods and data visualization techniques.

About Data

The Netflix Movies & TV Shows dataset can be found on Kaggle. It contains all TV Shows and Movies metadata available on Netflix. The dataset is updated every month. It contains 8807 records and 12 columns.

Netflix was founded on August 29, 1997, as a mail-based rental business. In January 2007, the company launched a streaming media service, introducing video on demand via the Internet.

Objectives

The purpose of this analysis is to answer the following questions:

1.) Number of movies/tv-shows added to the streaming platform by year.
2.) Which month has the most added movies/tv-shows?
3.) Which day has the most added movies/tv-shows?
4.) How many Movies vs. TV-Shows?
5.) Which year has the most releases Movies/TV-Shows?
6.) Which is the oldest movie/tv-show on streaming?
7.) Which countries produce the most movies/tv-shows?
8.) Cast members with the most content

# Phase-2 ": Data Preparation.

Import Labilaries

In [ ]:
import pandas as pd
import plotly.express as px

In [ ]:
df = pd.read_csv('netflix_titles.csv')

In [ ]:
df.head()

show_id  ...                                        description
0      s1  ...  As her father nears the end of his life, filmm...
1      s2  ...  After crossing paths at a party, a Cape Town t...
2      s3  ...  To protect his family from a powerful drug lor...
3      s4  ...  Feuds, flirtations and toilet talk go down amo...
4      s5  ...  In a city of coaching centers known to train I...

[5 rows x 12 columns]

In [ ]:
df.shape

(8807, 12)

In [ ]:
df.sample()

show_id  ...                                        description
3940   s3941  ...  After a tragedy at a school sends shock waves ...

[1 rows x 12 columns]

In [ ]:
len(df)

8807

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      8804 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB


In [ ]:
df.describe()

release_year
count   8807.000000
mean    2014.180198
std        8.819312
min     1925.000000
25%     2013.000000
50%     2017.000000
75%     2019.000000
max     2021.000000

## Process

Checking Missing Data

In [ ]:
missing_data = df.isna().sum().sort_values(ascending=False)
missing_data

director        2634
country          831
cast             825
date_added        10
rating             4
duration           3
description        0
listed_in          0
release_year       0
title              0
type               0
show_id            0
dtype: int64

In [ ]:
df_isna = pd.isna(df['director'])
df[df_isna]

show_id  ...                                        description
1         s2  ...  After crossing paths at a party, a Cape Town t...
3         s4  ...  Feuds, flirtations and toilet talk go down amo...
4         s5  ...  In a city of coaching centers known to train I...
10       s11  ...  Sicily boasts a bold "Anti-Mafia" coalition. B...
14       s15  ...  Cameras following Bengaluru police on the job ...
...      ...  ...                                                ...
8795   s8796  ...  Now that he's discovered the Pendulum Summonin...
8796   s8797  ...  During the Mongol invasions, Yunus Emre leaves...
8797   s8798  ...  Teen surfer Zak Storm is mysteriously transpor...
8800   s8801  ...  Strong-willed, middle-class Kashaf and carefre...
8803   s8804  ...  While living alone in a spooky town, a young g...

[2634 rows x 12 columns]

Checking for Duplicates

In [ ]:
df['show_id'].duplicated().any()

False

Changing the data format of columns date_added to datetime

In [ ]:
df['date_added'] = pd.to_datetime(df['date_added'].str.strip(), format= "%B %d, %Y")
df

show_id  ...                                        description
0         s1  ...  As her father nears the end of his life, filmm...
1         s2  ...  After crossing paths at a party, a Cape Town t...
2         s3  ...  To protect his family from a powerful drug lor...
3         s4  ...  Feuds, flirtations and toilet talk go down amo...
4         s5  ...  In a city of coaching centers known to train I...
...      ...  ...                                                ...
8802   s8803  ...  A political cartoonist, a crime reporter and a...
8803   s8804  ...  While living alone in a spooky town, a young g...
8804   s8805  ...  Looking to survive in a world taken over by zo...
8805   s8806  ...  Dragged from civilian life, a former superhero...
8806   s8807  ...  A scrappy but poor boy worms his way into a ty...

[8807 rows x 12 columns]

Checking uniques type of data of the column 'Type'
There are two type of data:Movies and TV Show

In [ ]:
data_type = df.type.unique()
data_type

array(['Movie', 'TV Show'], dtype=object)

Phase-4 Exploratory Data Analysis
About Netflix streaming platform

Number of Movies/TV-Shows added to the Streaming platform by Year

In [ ]:
# Year With the most added movies/TV-Show.
netflix_release_date = df.date_added.dt.year.astype('Int64').value_counts()
netflix_release_date

2019    2016
2020    1879
2018    1649
2021    1498
2017    1188
2016     429
2015      82
2014      24
2011      13
2013      11
2012       3
2008       2
2009       2
2010       1
Name: date_added, dtype: Int64

In [ ]:
# Month with most added movies/TV-Show
netflix_release_month = df.date_added.dt.month.astype('Int64').value_counts()
netflix_release_month

7     827
12    813
9     770
4     764
10    760
8     755
3     742
1     738
6     728
11    705
5     632
2     563
Name: date_added, dtype: Int64

In [ ]:
# Day with most day added movie/TV Show.
netflix_release_day = df.date_added.dt.day.astype('Int64').value_counts()
netflix_release_day

1     2212
15     687
2      325
16     289
31     274
20     249
19     243
5      231
22     230
10     214
30     210
6      210
18     207
26     206
8      201
14     198
25     197
27     195
7      194
21     193
28     190
23     184
12     181
17     180
13     175
4      175
24     159
3      151
11     149
9      147
29     141
Name: date_added, dtype: Int64

In [ ]:
# Number of Movies Vs TV Show.
netflix_type = df.type.value_counts()
netflix_type

Movie      6131
TV Show    2676
Name: type, dtype: int64

# About the Movies/TV Show.

In [ ]:
# The Year with the most release movies/TV-Show.
movietv_release_year =  df.release_year.value_counts() 
movietv_release_year

2018    1147
2017    1032
2019    1030
2020     953
2016     902
        ... 
1959       1
1961       1
1925       1
1947       1
1966       1
Name: release_year, Length: 74, dtype: int64

Oldest Movies/TV-Show on Streaming.

In [ ]:
df[df['release_year']== 1925]

show_id  ...                                        description
4250   s4251  ...  This collection restores films from women who ...

[1 rows x 12 columns]

Top 30 Country producing the most movies/TV-Show.


In [ ]:
# New dataset for the country count.
country_count = df.copy()
country_count = pd.concat([country_count, df['country'].str.split(",", expand=True)], axis=1)
country_count = country_count.melt(id_vars=["type","title"], value_vars=range(12), value_name="Country")
country_count = country_count[country_count["Country"].notna()]
country_count["Country"] = country_count["Country"].str.strip()
country_count

type                          title variable         Country
0         Movie           Dick Johnson Is Dead        0   United States
1       TV Show                  Blood & Water        0    South Africa
4       TV Show                   Kota Factory        0           India
7         Movie                        Sankofa        0   United States
8       TV Show  The Great British Baking Show        0  United Kingdom
...         ...                            ...      ...             ...
78859     Movie            The Look of Silence        8         Germany
85496     Movie                       Barbecue        9          Sweden
87666     Movie            The Look of Silence        9     Netherlands
94303     Movie                       Barbecue       10   United States
103110    Movie                       Barbecue       11         Uruguay

[10019 rows x 4 columns]

In [ ]:
# Country Unique.
country_count.Country.unique()   ### country should start capital letter i.e(Country)

array(['United States', 'South Africa', 'India', 'United Kingdom',
       'Germany', 'Mexico', 'Turkey', 'Australia', 'Finland', 'China',
       'Nigeria', 'Japan', 'Spain', 'France', 'Belgium', 'South Korea',
       'Argentina', 'Russia', 'Canada', 'Hong Kong', 'Italy', '',
       'Ireland', 'New Zealand', 'Jordan', 'Colombia', 'Switzerland',
       'Israel', 'Brazil', 'Taiwan', 'Bulgaria', 'Poland', 'Saudi Arabia',
       'Thailand', 'Indonesia', 'Egypt', 'Kuwait', 'Malaysia', 'Vietnam',
       'Sweden', 'Lebanon', 'Romania', 'Philippines', 'Iceland',
       'Denmark', 'United Arab Emirates', 'Netherlands', 'Norway',
       'Syria', 'Mauritius', 'Austria', 'Czech Republic', 'Cameroon',
       'Uruguay', 'Kenya', 'Chile', 'Luxembourg', 'Bangladesh',
       'Portugal', 'Hungary', 'Senegal', 'Singapore', 'Serbia', 'Namibia',
       'Peru', 'Mozambique', 'Belarus', 'Ghana', 'Zimbabwe',
       'Puerto Rico', 'Pakistan', 'Cyprus', 'Paraguay', 'Croatia',
       'Cambodia', 'Georgia', 'Sovie

In [ ]:
# Country with most number of content Streaming
country_count.Country.value_counts()

United States     3690
India             1046
United Kingdom     806
Canada             445
France             393
                  ... 
Palestine            1
Mozambique           1
East Germany         1
Latvia               1
Montenegro           1
Name: Country, Length: 123, dtype: int64

In [ ]:
# Top 30 Cast member with the most Contain.
cast_count = df.copy()
cast_count = pd.concat([cast_count, df['cast'].str.split(",",expand=True)],axis=1)
cast_count = cast_count.melt(id_vars=["type","title"], value_vars=range(44),value_name="Cast_name")
cast_count = cast_count[cast_count["Cast_name"].notna()]
cast_count["Cast_name"] = cast_count["Cast_name"].str.strip()
cast_count

type                             title variable        Cast_name
1       TV Show                     Blood & Water        0       Ama Qamata
2       TV Show                         Ganglands        0    Sami Bouajila
4       TV Show                      Kota Factory        0       Mayur More
5       TV Show                     Midnight Mass        0      Kate Siegel
6         Movie  My Little Pony: A New Generation        0  Vanessa Hudgens
...         ...                               ...      ...              ...
380555  TV Show                   Social Distance       43      Niles Fitch
382150  TV Show                       Creeped Out       43  Andonis Anthony
382475  TV Show                      Black Mirror       43   Michael Smiley
382921  TV Show            COMEDIANS of the world       43  Adib Alkhalidey
384887    Movie                  Arthur Christmas       43   Brian Cummings

[64105 rows x 4 columns]

In [ ]:
cast_count.Cast_name.value_counts()[:30]

Anupam Kher            43
Shah Rukh Khan         35
Julie Tejwani          33
Naseeruddin Shah       32
Takahiro Sakurai       32
Rupa Bhimani           31
Om Puri                30
Akshay Kumar           30
Yuki Kaji              29
Amitabh Bachchan       28
Paresh Rawal           28
Boman Irani            27
Vincent Tong           26
Rajesh Kava            26
Andrea Libman          25
Kareena Kapoor         25
Samuel L. Jackson      24
John Cleese            24
Fred Tatasciore        23
Jigna Bhardwaj         23
Tara Strong            23
Daisuke Ono            22
Nicolas Cage           21
Ashleigh Ball          21
Nawazuddin Siddiqui    21
Ajay Devgn             21
Junichi Suwabe         21
Kay Kay Menon          21
Salman Khan            20
Adam Sandler           20
Name: Cast_name, dtype: int64

# Phase-5: Visulazation

Movies Vs TV-Show

In [ ]:
px.histogram(df, x='type', color='type',
             title="Movies Vs.TV-Shows",
             color_discrete_sequence=px.colors.sequential.Sunsetdark)

In [ ]:
px.histogram(df, x= 'type', color= 'type',
             title="Movies vs. TV-Shows",
             color_discrete_sequence= px.colors.sequential.Sunsetdark)

Number of Movies/TV-Shows added by Year

In [ ]:
px.histogram(df, x=df['date_added'].dt.year, color=df['type'],
             title="Netflix number of Movies/TV Show by year",
             color_discrete_sequence=px.colors.sequential.Sunsetdark,
             labels=dict(x="Year", color="Type")
             )

# Number of Movies/TV-Shows added by Month

Dropping 'NA' records from the column 'date_added

Dropping 10 records from the column 'date_added' that contain 'NA' values

In [ ]:
# Counting the number of 'NA' on the column 'date_added'
df['date_added'].isna().sum()

10

In [ ]:
# Dropping 'NA'
df = df.dropna(subset=['date_added'])

In [ ]:
px.histogram(df, x=df['date_added'].dt.month, color=df
['type'],
                   color_discrete_sequence=px.colors.sequential.Sunsetdark,
                   title="Movies/TV Shows added by Month",
                   labels=dict(x="Month"))

Number of Movies/TV-Shows added by Day

In [ ]:
px.histogram(df, x= df['date_added'].dt.day,color= df['type'],
             color_discrete_sequence = px.colors.sequential.Sunsetdark,
             title="Movies/TV Shows added by day",
             labels=dict(x="day"))

Number of Movies/TV-Shows by year of release

In [ ]:
px.histogram(df, x= 'release_year', color='type',
             title="Number of Movies/TV-Shows by year of release",
             color_discrete_sequence= px.colors.sequential.Sunsetdark,
             labels={'release_year':'Year of release'}
                   )

In [ ]:
px.histogram(df, x= 'release_year', color='type',
             title="Number of Movies/TV-Shows by year of release",
             color_discrete_sequence= px.colors.sequential.Sunsetdark,  
             labels={'release_year':'Year of release'}                     
                   )

Top 30 Countries with the most streaming content

In [ ]:
px.histogram(country_count, x= 'Country', color= 'type',
        title="Top 30 Countries with the most streaming content",
        color_discrete_sequence= px.colors.sequential.Sunsetdark).update_xaxes(
        categoryorder="total descending",range=(0, 30))

Top 30 Cast members with the most titles

In [ ]:
px.histogram(cast_count, x= 'Cast_name', color= 'type',
    title="Top 30 Cast members with the most streaming content",
    color_discrete_sequence= px.colors.sequential.Sunsetdark).update_xaxes(
    categoryorder="total descending",range=(0, 30))

# Findings

About Netflix

1. There are more Movies than TV-Shows available on streaming. 6131 movies and 2676 tv-shows.

2. 2019 is the year with the most content addition on the streaming platform, 2016 movie/tv-shows added, followed by 2020 with 1879, and 2018 with 1649 total

3. July and December are the months with the most content addition, 827 and 813 movie/tv-shows added.

4. Netflix adds content on the first day of the month more than any other day.

About the content available

1. Among the contents available 1147 of them were originally released in 2018 followed by 2017 with 1032, and 2019 with 1030 total.

2. Pioneers: First Women Filmmakers is the oldest content available on streaming. It's a collection of restored films dating from 1925.

3. The United States is the country that produces the most of the content with 3690 titles, followed by India 1046 titles and the United Kingdom 806 titles.

4. Anupam Kher is the actor with the higher number of titles, 43 films. Anupam Kher is an Indian actor, director, and producer that has appeared in over 500 films.